### ISSUE TITLE
Problems with redis after 22.11.0 update

### ISSUE THREAD

**Issue by hheexx** [getsentry/self-hosted#1844]
### Self-Hosted Version

22.11.0

### CPU Architecture

x86_64

### Docker Version

20.10.21

### Docker Compose Version

1.29.2

### Steps to Reproduce

After upgrade to 12.11 from 12.9 we receive strange redis errors in Internal. After 1 day. 
Some time after that sentry stoped working.

### Expected Result

no errors

### Actual Result

consumer: Cannot connect to redis://redis:6379/0: BusyLoadingError('Redis is loading the dataset in memory').
Trying again in 8.00 seconds... (4/100)

### Event ID

a532e3b34bc842a083e26b65c76a60a8

**Comment by hheexx** [comment]
It may be RAM related.
Looks like 16GB is not enough any more under load.

**Comment by hubertdeng123** [comment]
Hmmm, got it. Did increasing the RAM resolve this issue?

**Comment by hubertdeng123** [comment]
I'm wondering if it's reasonable to revisit the minimum requirements to run self-hosted Sentry

**Comment by hheexx** [comment]
> Hmmm, got it. Did increasing the RAM resolve this issue?

I'm not 100% sure yet. It still has problems with loading issues but it looks like internal errors are solved.
I incresed from 16GB to 20GB.

**Comment by hheexx** [comment]
I still receive errors. It looks memory is a problem but not event 20GB is enough. Will now increse to 24GB.
It may be memory leak.

![image](https://user-images.githubusercontent.com/103767/205763043-cee01f46-39d9-4056-817e-d866686957c9.png)

**Comment by hheexx** [comment]
oom-killler is killing it 
`Dec 05 15:11:14 sentry kernel: Out of memory: Killed process 4127921 (redis-server) total-vm:13695552kB, anon-rss:11567492kB, file-rss:0kB, shmem-rss:0kB, UID:999 pgtables:23960kB oom_score_adj:0`

**Comment by igorcoding** [comment]
Seems like the same problem as in https://github.com/getsentry/self-hosted/issues/1787 and https://github.com/getsentry/self-hosted/issues/1830. It is really a weird issue and as I can see there is still no solution found :(

**Comment by hheexx** [comment]
But disk IO is still 4x larger then before upgrade

**Comment by pavels** [comment]
Maybe this will help, maybe not - but here is what we found

Our sentry instance is currently processing something between 15k-20k events per minute. Version 22.11 What i found is that whenever there is backlog in event processing, redis will grow VERY FAST - because i have memory limit set and no eviction, after it reaches full memory, it will stop taking more data an everything stops and the backlog will start to grow rapidly as well - so clearing redis will help only for some time and it will fill up again.

Eviction as suggested by some will only sidestep this problem. I checked now and our redis now takes about 30MB of memory.

There multiple parts involved
1) consumers (sentry and snube) - basically event processing pipeline
2) workers
3) postgresql database

Any slightly wrong setting and with our event rate and backlog goes high and redis goes quickly out of memory.

`mach_batch_size` in consumers matters a lot - for us 256 works well, but it is higlhy dependent on infrastructure
Also check if your workes are not overloaded and that your postgres is coping wit the load.

Basically you should somehow monitor your event processing backlog, and you you can't process fast enough, find the problem and solve it. That is what worked for us.

Also if you have millions of events in your retention period, the default `node_store` (stored as blobs in postgresql table) is useless.

**Comment by hheexx** [comment]
We are at 100 events per minute, wayy less then you.
But our events have relatively large breadcrumbs.
HW is 8 core VM on Ryzen 5950X

**Comment by hheexx** [comment]
btw. number I mentioned for epm is actually for number of errors. We have a lot more transactions but i'm not sure how to check how meny. I'm looking at Stats tab now and it shows all incoming transactions are filtered and it shows arround 10K/m. But as I see transaction data is OK in projects and also we don't have any special incoming filters, just known web crawlers selected.

**Comment by pavels** [comment]
By events i mean transactions and errors together - transactions for performance monitoring also goes through redis

**Comment by hheexx** [comment]
ingest:

```
17:43:37 [INFO] batching-kafka-consumer: Worker flush took 87ms

17:43:38 [INFO] batching-kafka-consumer: Flushing 20 items (from {('ingest-transactions', 0): [325876898, 325876917]}): forced:False size:False time:True

17:43:38 [INFO] batching-kafka-consumer: Worker flush took 75ms

17:43:40 [INFO] batching-kafka-consumer: Flushing 16 items (from {('ingest-transactions', 0): [325876918, 325876932], ('ingest-events', 0): [101407812, 101407812]}): forced:False size:False time:True

17:43:40 [INFO] batching-kafka-consumer: Worker flush took 76ms

17:43:41 [INFO] batching-kafka-consumer: Flushing 18 items (from {('ingest-transactions', 0): [325876933, 325876949], ('ingest-events', 0): [101407813, 101407813]}): forced:False size:False time:True

17:43:41 [INFO] batching-kafka-consumer: Worker flush took 83ms

17:43:42 [INFO] batching-kafka-consumer: Flushing 17 items (from {('ingest-transactions', 0): [325876950, 325876966]}): forced:False size:False time:True

17:43:42 [INFO] batching-kafka-consumer: Worker flush took 92ms


```




worker:
```
17:28:50 [WARNING] sentry.tasks.release_registry: Release registry URL is not specified, skipping the task.

17:29:57 [WARNING] sentry.http: source.disabled (type='too_large_for_cache' url='https://crm.cx.rs/app/__dist/main.js.map')

17:33:50 [WARNING] sentry.tasks.release_registry: Release registry URL is not specified, skipping the task.

17:37:23 [WARNING] sentry.http: source.disabled (type='fetch_generic_error' value="<class 'requests.exceptions.ConnectionError'>" url='https://localhost:44333/app/__dist/main.js?v=1.0.0.0')

17:38:50 [WARNING] sentry.tasks.release_registry: Release registry URL is not specified, skipping the task.

17:39:41 [WARNING] sentry.http: source.disabled (type='too_large_for_cache' url='https://crm.cx.rs/app/__dist/main.js.map')

17:43:38 [INFO] sentry.tasks.update_user_reports: update_user_reports.records_updated (reports_to_update=0 reports_with_event=0 updated_reports=0)

17:43:50 [WARNING] sentry.tasks.release_registry: Release registry URL is not specified, skipping the task.

17:44:44 [WARNING] sentry.http: source.disabled (type='too_large_for_cache' url='https://crm.cx.rs/app/__dist/main.js.map')
```

**Comment by hheexx** [comment]
![image](https://user-images.githubusercontent.com/103767/207117809-55cad8c8-dbe5-42bb-8545-c70076662ac2.png)
![image](https://user-images.githubusercontent.com/103767/207117896-6ae06787-f388-4ccc-beca-f53235123588.png)
![image](https://user-images.githubusercontent.com/103767/207118041-d874b37d-0cb8-447d-a556-c9f5422a0dbd.png)

**Comment by pavels** [comment]
Try to run `sentry queues list` in `web` container and look for large numbers of jobs

**Comment by hheexx** [comment]
looks like 22.12 fixed both redis and stats
